cli/src/commands/help.rs

```diff
@@ -28,22 +32,98 @@ pub(crate) struct HelpArgs {
 
 #[instrument(skip_all)]
 pub(crate) fn cmd_help(
-    _ui: &mut Ui,
+    ui: &mut Ui,
     command: &CommandHelper,
     args: &HelpArgs,
 ) -> Result<(), CommandError> {
+    // TODO: Add all documentation to categories
+    //
+    // Maybe adding some code to build.rs to find all the docs files and build the
+    // BTreeMap at compile time.
+    //
+    // It would be cool to follow the docs hierarchy somehow.
+    //
+    // One of the problems would be `config.md`, as it has the same name as a
+    // subcommand.
+    let categories = BTreeMap::from([
+        (
+            "revsets",
+            Category {
+                description: "A functional language for selecting a set of revision",
+                content: include_str!("../../../docs/revsets.md"),
```

Comment:
I had hoped that we would be able to render the markdown in the terminal using ANSI escape codes somehow. I guess the source is fine to start with. Add a TODO about rendering it?

It's unfortunate that there are so many flavors of markdown and that we'll have to find something that's able to format the same markdown source as MkDocs is able to format for the website.

(Related to @thoughtpolice's PR-level comment)

Reply:
OT: the markdown renderer could also be a template function so that user can do `markdown(description)`.

Reply:
Interesting. So the raw template output (before formatting) would include color codes then? That seems a bit weird, but I guess it works (we don't escape escapes in the formatter input, do we?).

Reply:
We do escapes. I assumed the renderer would emit labels, not ANSI escapes.

Reply:
Ah, of course. Good idea.

Reply:
Added a TODO for that, referencing #3130. I think that the work of @thoughtpolice has some great ideas. As he said, I also think that might be a future PR improving this with some of his ideas.